tuning/sft_trainer.py

```diff
@@ -360,7 +361,11 @@ def train(
         for x in framework.get_callbacks_and_ready_for_train(model, accelerator):
             trainer.add_callback(x)
 
-    trainer.train()
+    # Check if tuning has to be resumed from last saved checkpoint
+    is_checkpoint_available = get_last_checkpoint(training_args.output_dir)
+    if is_checkpoint_available:
+        logger.info("Tuning resumes from the last checkpoint")
+    trainer.train(resume_from_checkpoint=is_checkpoint_available)
```

Comment:
`resume_from_checkpoint=True` gives same effect isn't it?

Reply:
`resume_from_checkpoint=True`, Yes I tested it. It gives the same effect if the `output_dir` has checkpoints available or else throws error.

Reply:
It would be helpful for me to know your requirements on default flow when user does not pass the flag as then could be able to finalize the code changes for review.

Reply:
@ashokponkumar Any comment/input on above ?

Reply:
What error do we get when `resume_from_checkpoint` is set to True and there is no checkpoint? We can go with this fix for now, but it would be good to check in the HF Trainer, if they are okay with gracefully failing. Even if `resume_from_checkpoint` is set, if there is no checkpoint in the output dir, it should start like a normal run from scratch.

Reply:
Also, if user actually provides a false explicitly we should honor it.

Reply:
>What error do we get when resume_from_checkpoint is set to True and there is no checkpoint

Error output and failing of code and exiting.
`ValueError: No valid checkpoint found in output directory (outputs/lora-tuning)`

Reply:
>Even if resume_from_checkpoint is set, if there is no checkpoint in the output dir, it should start like a normal run from scratch.

>Also, if user actually provides a false explicitly we should honor it.

Okay.

Reply:
Based on above comments, I have pushed few changes.
@kmehant Feel free to review it. 

Cases covered and tested:

1- **If user passes the flag `resume_from_checkpoint` with value** `False`: Tuning will start from scratch.

2- **If user passes the flag `resume_from_checkpoint` with value** `True`: if `output_dir` has checkpoints, then tuning will resume from last checkpoint, else tuning will start from scratch.

3- **If user passes the flag `resume_from_checkpoint` with value** `String: Entire Path of checkpoint-x`: The model would be loaded from `Path of checkpoint-x` and tuning will resume from the `checkpoint-x`.

4- **If user doesn't pass flag `resume_from_checkpoint`** : if `output_dir` has checkpoints, then tuning will resume from last checkpoint, else tuning will start from scratch.